### PR TITLE
[Encode] code alignment

### PR DESCRIPTION
**Pull request by lwan108**
Signed-off-by: LijuanWang <lijuan.wang@intel.com>

### CODE REVIEW

**Review by dmitryermilov — APPROVED** [review]

**Review by vilichev — DISMISSED** [review]

**Review by thomasli21801 — DISMISSED** [review]

**Review by ansokolo — APPROVED** [review]

File: _studio/enctools/src/mfx_enctools_lpla.cpp [hunk 1]
```diff
@@ -241,6 +241,7 @@ mfxStatus LPLA_EncTool::ConfigureExtBuffs(mfxEncToolsCtrl const & ctrl, mfxExtEn
         m_extBufLPLA.LookAheadScaleX = m_extBufLPLA.LookAheadScaleY = (mfxU8)m_lookAheadScale;
 
         m_extBufLPLA.GopRefDist = ctrl.MaxGopRefDist;
+        m_extBufLPLA.codecTypeInEncodePass = ctrl.CodecId;
```
Comment (takulako) [inline _studio/enctools/src/mfx_enctools_lpla.cpp:244]: codecTypeInEncodePass  is currently missing in the OS version of mfxExtLplaParam struct  in mfx_ext_buffers.h.  I guess we'd better add it there to keep the code compilable with MFX_ENABLE_ENCTOOLS_LPLA defined.

File: _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp [hunk 2]
```diff
@@ -1471,7 +1471,7 @@ namespace MPEG2EncoderHW
             mfxFrameAllocRequest request = {};
             sts = QueryIOSurf(m_pCore, par, &request);
             MFX_CHECK(sts>=0,sts);
-            m_VideoParamsEx.encNumFrameMin = request.NumFrameMin;
+
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp:1474]: are we sure we can delete this?

**Review by mgonchar — CHANGES_REQUESTED** [review]

File: _studio/enctools/src/mfx_enctools_lpla.cpp (hunk 1 above)
Comment (lwan108) [reply _studio/enctools/src/mfx_enctools_lpla.cpp:244]: Thanks Timofei. Adding codecTypeInEncodePass in mfxExtLplaParam struct will be together with LPLA enabling on linux.

File: _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h [hunk 3]
```diff
@@ -2646,8 +2647,6 @@ class H264EncTools
     class DdiTask2ndField
     {
     public:
```
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h:2649]: It's to align with latest close source code. @takulako , could you please comment which one is expected?

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 4]
```diff
@@ -2257,13 +2255,6 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
             par.mfx.GopRefDist = 1;
         }
```
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp:2257]: It's to align with latest close source code. Need to add this to close source code, if this is useful.

File: _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp (hunk 2 above)
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp:1474]: It's to align with latest close source code, no such code in CS.

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp [hunk 5]
```diff
@@ -4682,7 +4680,6 @@ mfxStatus ImplementationAvc::QueryStatus(
     mfxU32    fid,
     bool      useEvent)
 {
```
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp:4682]: To align with CS.

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp:2257]: To align with CS.

File: _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (takulako) [reply _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h:2649]: The line was added in OS by @onabiull in April'21
SHA-1: 1a55c256657728fa76244134cc304fafd73ab91d
* Initial oneVPL GPU Runtime upstream

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp:4682]: CS code was removed @4365da4, PR [1844](https://github.com/intel-innersource/drivers.gpu.media.sdk.lib/pull/1844 ).
@dmitryermilov

File: _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h:2649]: Thanks@takulako. Reverted OS removal, and added this in CS. @DenWolf pls comment if the change isn't expected.

File: _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp (hunk 2 above)
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp:1474]: CS SHA :0713bc98fe28124debcd221037878ecf15ebc67f, PR[1723](https://github.com/intel-innersource/drivers.gpu.media.sdk.lib/pull/1723), @AntonGrishin

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp:2257]: CS SHA:2a2ac49e40472478e666082e6c3d1202affd43e4, PR[#1264](https://github.com/intel-innersource/drivers.gpu.media.sdk.lib/issues/1264),@saosipov

File: _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (lwan108) [reply _studio/mfx_lib/encode_hw/h264/include/mfx_h264_encode_hw_utils.h]: CS PR:[#2548](https://github.com/intel-innersource/drivers.gpu.media.sdk.lib/pull/2548)

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (mgonchar) [reply _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp:4682]: ok, thanks

File: _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp (hunk 2 above)
Comment (mgonchar) [reply _studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp:1474]: thanks for clarification!

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (mgonchar) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp:2257]: got it, thanks

**Review by mgonchar — APPROVED** [review]

**Review by thomasli21801 — APPROVED** [review]

**Review by dmitryermilov — APPROVED** [review]

**Review by ansokolo — APPROVED** [review]

**Review by takulako — APPROVED** [review]

**Review by ogolikov — APPROVED** [review]

**Pull request merged**